hypervisor/src/kvm/mod.rs

```diff
@@ -937,19 +939,29 @@ impl vm::Vm for KvmVm {
     /// Initialize TDX for this VM
     ///
     #[cfg(feature = "tdx")]
-    fn tdx_init(&self, cpuid: &[CpuIdEntry], max_vcpus: u32) -> vm::Result<()> {
+    fn tdx_init(
+        &self,
+        cpuid: &[CpuIdEntry],
+        max_vcpus: u32,
+        mrconfigid_opt: &Option<String>,
+    ) -> vm::Result<()> {
         const TDX_ATTR_SEPT_VE_DISABLE: usize = 28;
 
         let mut cpuid: Vec<kvm_bindings::kvm_cpuid_entry2> =
             cpuid.iter().map(|e| (*e).into()).collect();
         cpuid.resize(256, kvm_bindings::kvm_cpuid_entry2::default());
 
+        let mut mrconfigid = [0u8; 48];
+        if let Some(mrconfigid_str) = mrconfigid_opt {
+            decode_to_slice(mrconfigid_str, &mut mrconfigid as &mut [u8]).unwrap();
+        }
+
         #[repr(C)]
         struct TdxInitVm {
             attributes: u64,
             max_vcpus: u32,
             padding: u32,
-            mrconfigid: [u64; 6],
+            mrconfigid: [u8; 48],
```

Comment:
Does it indicate you are working on a different version of host kernel that comes with a different `TdxInitVm` struct?